shellfuncs/core.py

```diff
@@ -66,15 +66,25 @@ class ShellScriptFinder:
     """
     @classmethod
     def find_spec(cls, name, path=None, target=None):
-        script_path = Path('{0}.sh'.format(name.replace('.', '/')))
-        if not script_path.exists():  # no such script found.
-            return None
-
-        # TODO: check relative imports and directory walking
-        loader = ShellScriptLoader()
-        spec = importlib.util.spec_from_loader(name, loader)
-        spec.script = script_path
-        return spec
+        # import is of the form: `from path.to.script import`
+        if len(name.split('.')) > 1:
+            # need to convert any . to / for path finding
+            script_path = Path('{0}.sh'.format(name.replace('.', '/')))
+        else:
+            script_path = name + '.sh'
+
+        # look for requested script in python path
+        for p in sys.path:
+            # convert script name to abs path
+            place = str(os.path.join(p, script_path))
+            if os.path.exists(place):
```

Comment:
```suggestion
            if place.exists():
```

Maybe we should check if it's really a script and not just a path which exists ?

Reply:
Agreed, I have the luxury in my environment of making assumptions, but this should definitely be robustified for general use.

Is there a good way you can think of beyond checks for execute privileges, '.sh' file extension verification, reading the first line of the file for a shebang? Those are the ideas off the top of my head before I dig in to investigate